### PR TITLE
[WIP] Implement XDG Base Directory Spec support

### PR DESCRIPTION
**Pull request by ammgws**
Solves #57

Need to think about the local/host xxh_home default directory as well...

### CODE REVIEW

File: xxh_xxh/xxh.py [hunk 1]
```diff
@@ -70,6 +70,13 @@ def eprint(self, *args, **kwargs):
     def eeprint(self, *args, **kwargs):
         eeprint(*args, **kwargs)
 
+    @staticmethod
+    def generate_config_filepath():
+        if os.environ.get('XDG_CONFIG_HOME'):
+            return p(os.environ['XDG_CONFIG_HOME']) / 'xxh' / 'xxhc'
+        else:
+            return p(os.environ['HOME']) / 'xxh' / 'xxhc'
```
Comment (snowe2010) [inline xxh_xxh/xxh.py]: this should default to `~/.config` if it is to follow the [XDG spec](https://wiki.archlinux.org/index.php/XDG_Base_Directory).

Comment (ammgws) [reply xxh_xxh/xxh.py]: $HOME should be OK. From the actual spec:

> $XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

Comment (snowe2010) [reply xxh_xxh/xxh.py]: ... which states that the default should be `~/.config`, not `~`

Comment (ammgws) [reply xxh_xxh/xxh.py]: @snowe2010 Ah, I didn't notice I had messed up the code. Cheers

**Pull request merged**